### ISSUE TITLE
axios uses http adaptor when https_proxy is set

### ISSUE THREAD

**Issue by PavelPolyakov** [axios/axios#1885]
First I stumbled upon the socket hangs up issue. So I found this PR, which was merged, but never released: https://github.com/axios/axios/pull/959 .

So I installed `axios@next` to check if this issue is fixed there. And found another problem.

When `http_proxy` and `https_proxy` is set, and in my case it's `socks5://` proxy - `axios` can not get the url.

This is how I set proxy:
```javascript
export http_proxy=socks5://localhost:9999/ https_proxy=socks5://localhost:9999/
```

This is the code:
```javascript
const axios = require('axios');
const https = require('https');

const httpsAgent = new https.Agent({
  rejectUnauthorized: false,
});

async function main() {
  const response = await axios.get('https://api.jsonbin.io/b/5bf56cf6bc03835906819ee8', { httpsAgent });
  console.log(response.data);
}

main();
```

And here is the error:
```
(node:18469) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 3): Error: Protocol "http:" not supported. Expected "https:"
```

I'm sure it's about the proxy, since when I unset `https_proxy` - everything works.
It may be the issue, that I use `socks5://` as protocol, but I'm not sure.

#### Context

- axios version: *axios@0.19.0-beta.1*
- Environment: *v8.9.4*

**Comment by mapleeit** [comment]
I think we should discuss about is it the right way to **use HTTP_PROXY or HTTPS_PROXY by default without providing an option.**

In addition, it just uses http or https proxy. **What if I have a NO_PROXY**?

ref: #1514

**Comment by chinesedfan** [comment]
Closed in favor of #1981.
